osu.Game/Beatmaps/BeatmapInfoExtensions.cs

```diff
@@ -29,10 +29,29 @@ public static RomanisableString GetDisplayTitleRomanisable(this IBeatmapInfo bea
             return new RomanisableString($"{metadata.GetPreferred(true)}".Trim(), $"{metadata.GetPreferred(false)}".Trim());
         }
 
-        public static string[] GetSearchableTerms(this IBeatmapInfo beatmapInfo) => new[]
+        public static ReadOnlySpan<string> GetSearchableTerms(this IBeatmapInfo beatmapInfo)
         {
-            beatmapInfo.DifficultyName
-        }.Concat(beatmapInfo.Metadata.GetSearchableTerms()).Where(s => !string.IsNullOrEmpty(s)).ToArray();
+            Span<string> terms = new string[8];
+            int i = 0;
+            if (!string.IsNullOrEmpty(beatmapInfo.DifficultyName))
+                terms[i++] = beatmapInfo.DifficultyName;
+            var metadata = beatmapInfo.Metadata;
+            if (!string.IsNullOrEmpty(metadata.Author.Username))
+                terms[i++] = metadata.Author.Username;
+            if (!string.IsNullOrEmpty(metadata.Artist))
+                terms[i++] = metadata.Artist;
+            if (!string.IsNullOrEmpty(metadata.ArtistUnicode))
+                terms[i++] = metadata.ArtistUnicode;
+            if (!string.IsNullOrEmpty(metadata.Title))
+                terms[i++] = metadata.Title;
+            if (!string.IsNullOrEmpty(metadata.TitleUnicode))
+                terms[i++] = metadata.TitleUnicode;
+            if (!string.IsNullOrEmpty(metadata.Source))
+                terms[i++] = metadata.Source;
+            if (!string.IsNullOrEmpty(metadata.Tags))
+                terms[i++] = metadata.Tags;
+            return terms[..i];
```

Comment:
What is this range indexer even doing? It seems to be slicing an eight-element span... to eight elements?

Reply:
if any of the if statements fails, then `i` will be smaller

Reply:
There are many less weird ways to achieve this. Example: create a `new List<string>(8)`, add to it as normal, and then `.AsSpan()` it at the end. I believe it should be equivalent in performance and look a lot less weird.